code/modules/mob/living/simple_animal/hostile/megafauna/ancient_robot.dm

```diff
@@ -0,0 +1,667 @@
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/ex_act(severity, target)
+	if(severity == EXPLODE_LIGHT)
+		return
+	..()
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/Goto()
+	if(!exploding)
+		return ..()
+	return
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/Moved(atom/OldLoc, Dir, Forced = FALSE)
+	if(charging)
+		DestroySurroundings()
+	playsound(src, 'sound/effects/meteorimpact.ogg', 200, TRUE, 2, TRUE)
+	if(Dir)
+		leg_walking_controler(Dir)
+		if(charging)
+			if(mode == PYRO)
+				var/turf/C = get_turf(src)
+				new /obj/effect/temp_visual/lava_warning(C)
+				for(var/turf/T in range (1,src))
+					new /obj/effect/hotspot(T)
+					T.hotspot_expose(700,50,1)
+			if(mode == VORTEX)
+				var/turf/C = get_turf(src)
+				C.ex_act(3)
+
+	beam.forceMove(get_turf(src))
+	return ..()
+
+/mob/living/simple_animal/hostile/ancient_robot_leg
+	name = "leg"
+	desc = "leg"
+	icon = 'icons/mob/lavaland/lavaland_monsters.dmi'
+	icon_state = "leg"
+	maxHealth = INFINITY //it's fine trust me
+	health = INFINITY
+	faction = list("mining", "boss") // No attacking your leg
+	weather_immunities = list("lava","ash")
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	flying = TRUE
+	check_friendly_fire = 1
+	ranged = TRUE
+	projectilesound = 'sound/weapons/gunshots/gunshot.ogg'
+	projectiletype = /obj/item/projectile/ancient_robot_bullet
+	attacktext = "stomps on"
+	armour_penetration = 40
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	obj_damage = 400
+	move_force = MOVE_FORCE_OVERPOWERING
+	move_resist = MOVE_FORCE_OVERPOWERING
+	pull_force = MOVE_FORCE_OVERPOWERING
+	sentience_type = SENTIENCE_BOSS
+	environment_smash = ENVIRONMENT_SMASH_RWALLS
+	robust_searching = TRUE
+	ranged_ignores_vision = TRUE
+	stat_attack = DEAD
+	var/range = 3
+	var/mob/living/simple_animal/hostile/megafauna/ancient_robot/core = null
+	var/fake_max_hp = 400
+	var/fake_hp = 400
+	var/fake_hp_regen = 10
+	var/transfer_rate = 0.75
+	var/who_am_i = null
+	var/datum/beam/leg_part
+
+/mob/living/simple_animal/hostile/ancient_robot_leg/Initialize(mapload, mob/living/ancient, who)
+	. = ..()
+	if(!ancient)
+		qdel(src) //no
+	core = ancient
+	who_am_i = who
+	ranged_cooldown_time = (rand(30, 60)) // keeps them not running on the same time
+	addtimer(CALLBACK(src, .proc/beam_setup), 1 SECONDS)
+
+/mob/living/simple_animal/hostile/ancient_robot_leg/Life(seconds, times_fired)
+	..()
+	health_and_snap_check(TRUE)
+
+/mob/living/simple_animal/hostile/ancient_robot_leg/Goto()
+	return // stops the legs from trying to move on their own
+
+/mob/living/simple_animal/hostile/ancient_robot_leg/proc/beam_setup()
+	leg_part = Beam(core.beam, "leg_connection", 'icons/effects/effects.dmi', time=INFINITY, maxdistance=INFINITY, beam_type=/obj/effect/ebeam)
+
+/mob/living/simple_animal/hostile/ancient_robot_leg/adjustHealth(amount, updating_health = TRUE)
```

Comment:
Maybe use a similar proc on the main robot to ensure it cant actually be "killed" unless forced (someone could deal 200 damage in the explode period, you never know)

Reply:
Done!